Packs/AWS-SNS-Listener/Integrations/AWSSNSListener/AWSSNSListener.py

```diff
@@ -0,0 +1,266 @@
+from CommonServerPython import *  # noqa: F401
+from CommonServerUserPython import *
+from tempfile import NamedTemporaryFile
+from traceback import format_exc
+from collections import deque
+import uvicorn
+from secrets import compare_digest
+from fastapi import Depends, FastAPI, Request, Response, status
+from fastapi.security import HTTPBasic, HTTPBasicCredentials
+from fastapi.security.api_key import APIKeyHeader
+from fastapi.openapi.models import APIKey
+import requests
+import base64
+from M2Crypto import X509
+
+
+PARAMS: dict = demisto.params()
+sample_events_to_store = deque(maxlen=20)  # type: ignore[var-annotated]
+
+app = FastAPI(docs_url=None, redoc_url=None, openapi_url=None)
+basic_auth = HTTPBasic(auto_error=False)
+token_auth = APIKeyHeader(auto_error=False, name='Authorization')
+
+
+def handle_proxy_and_ssl():
+    proxies = None
+    use_ssl = not PARAMS.get('insecure', False)
+    if not is_demisto_version_ge('8.0.0'):
+        return proxies, use_ssl
+    CRTX_HTTP_PROXY = os.environ.get('CRTX_HTTP_PROXY', None)
+    demisto.debug(f"CRTX_HTTP_PROXY: {CRTX_HTTP_PROXY}")
+    if CRTX_HTTP_PROXY:
```

Comment:
We should also take into consideration the normal proxy usage (could be relevant when CRTX_HTTP_PROXY is missing, but a proxy is configured).
For example, this configuration will not work with an engine that's configured with a proxy, or with an XSOAR 6 instance that uses proxy.